test/integration/appTemplates.spec.ts

```diff
@@ -28,21 +29,17 @@ test('can use images app template', async ({ page }) => {
   const runtimeModel = new ToolpadRuntime(page);
   await runtimeModel.gotoPage(app.id, 'dogBreedsPage');
 
-  const waitForDogAPIResponse = () => page.waitForResponse('https://images.dog.ceo/**');
-
-  await waitForDogAPIResponse();
-
   await page.locator('h3:has-text("Dog Images")').waitFor({ state: 'visible' });
 
   const breedInputLocator = page.locator('[aria-haspopup="listbox"]').first();
   await breedInputLocator.click();
   await page.locator('li:has-text("australian")').click();
-  await waitForDogAPIResponse();
+  await waitForDataResponse(page, baseURL!);
 
   const subBreedInputLocator = page.locator('[aria-haspopup="listbox"]').nth(1);
   await subBreedInputLocator.click();
   await page.locator('li:has-text("shepherd")').click();
-  await waitForDogAPIResponse();
+  await waitForDataResponse(page, baseURL!);
```

Comment:
These are implementation details. Why can't we just wait until the image has the expected attribute?

Reply:
Is there a way? I looked in the docs but couldn't find it.

Reply:
Not sure why I didn't try this one: `await expect(locator).toHaveAttribute('type', 'text');`
Will try it next, maybe it will work.

Reply:
> Is there a way?

By using a locator with an [attribute selector](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors)?